src/ui/public/utils/brush_event.js

```diff
@@ -3,14 +3,22 @@ import moment from 'moment';
 import { buildRangeFilter } from 'ui/filter_manager/lib/range';
 
 export function UtilsBrushEventProvider(timefilter) {
+  const RANGE_FILTER = 'range';
+  const TIME_FILTER = 'timefilter';
   return $state => {
     return event => {
       if (!event.data.xAxisField) {
         return;
       }
 
-      switch (event.data.xAxisField.type) {
-        case 'date':
+      let filterType = RANGE_FILTER;
+      if (event.data.xAxisField.type === 'date' &&
+        event.data.xAxisField.name === event.data.indexPattern.timeFieldName) {
+        filterType = TIME_FILTER;
```

Comment:
Is the `switch` really necessary? What if we just moved the block from the switch statement into the body of this `if` statement? Seems like it could simplify this logic a bit.